apps/omg_watcher/test/omg_watcher/integration/block_getter_1_test.exs

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.Integration.BlockGetter1Test do
+  @moduledoc """
+  This test is intended to be the major smoke/integration test of the Watcher
+
+  It tests whether valid/invalid blocks, deposits and exits are tracked correctly within the Watcher
+  """
+
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  use OMG.Fixtures
+  use OMG.Watcher.Integration.Fixtures
+  use Plug.Test
+
+  require OMG.Utxo
+
+  alias OMG.Eth
+  alias OMG.Watcher.BlockGetter
+  alias OMG.Watcher.Event
+  alias OMG.Watcher.Integration.BadChildChainServer
+  alias OMG.Watcher.Integration.TestHelper, as: IntegrationTest
+  alias Support.DevHelper
+  alias Support.RootChainHelper
+  alias Support.WatcherHelper
+
+  @timeout 40_000
+  @eth OMG.Eth.zero_address()
+
+  @moduletag :mix_based_child_chain
+
+  @moduletag timeout: 150_000
+
+  @tag fixtures: [:in_beam_watcher, :stable_alice, :token, :stable_alice_deposits, :test_server]
+  test "transaction which is spending an exiting output before the `sla_margin` causes an invalid_exit event only",
+       %{stable_alice: alice, stable_alice_deposits: {deposit_blknum, _}, test_server: context} do
+    Process.sleep(12_000)
```

Comment:
Yes, not optimal to put these all around, but as said, geth and childchain in docker are considerably slower so that amounts of wait times I do not want to address right now, since these tests belong in cabbage!